packages/server/modules/workspaces/tests/unit/services/workspaceProjectRoleCreation.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { StreamAclRecord, StreamRecord } from '@/modules/core/helpers/types'
+import { WorkspaceAcl } from '@/modules/workspacesCore/domain/types'
+import { grantWorkspaceProjectRolesFactory } from '@/modules/workspaces/services/workspaceProjectRoleCreation'
+import { Roles } from '@speckle/shared'
+import { expect } from 'chai'
+import cryptoRandomString from 'crypto-random-string'
+
+describe('Workspace project role services', () => {
+  describe('grantWorkspaceProjectRolesFactory creates a function, that', () => {
```

Comment:
Do we really need the test files to be this granular? There is an overheard to each test file - initializing a new test context, running `beforeEachContext` etc.

Some if not all of these can probably be merged into the same test file

Reply:
Ah, this is a unit test, so its probably lightweight enough to stand on its own. That said, I'd still evaluate if some of the tests can't be merged just to avoid the overheard there is and also be able to reuse more of test setup, without having to duplicate it

Reply:
Its a unittest, being verbose in the test context is actually good.

Reply:
will push back on this one - I think the verbosity/overhead is worth it here, but will keep this in mind going forwards